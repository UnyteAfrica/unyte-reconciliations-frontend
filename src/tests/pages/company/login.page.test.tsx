import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import React from "react";
import { CompanyLoginPage } from "@/pages/company/login.page";

function setup(reactNode: React.ReactNode) {
  return {
    user: userEvent.setup(),
    ...render(reactNode),
  };
}

describe("Company Login Page", () => {
  it("renders correctly", () => {
    const tree = render(
      <BrowserRouter>
        <CompanyLoginPage />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should validate company ID", async () => {
    const { user } = setup(
      <BrowserRouter>
        <CompanyLoginPage />
      </BrowserRouter>
    );

    const companyIdInput = screen.getByLabelText("Admin Email");
    await user.click(companyIdInput);
    await user.keyboard("john@");
    const submitButton = screen.getByText("Sign In");
    await user.click(submitButton);
    const errorMessage = screen.getByText("Admin Email is invalid");
    expect(errorMessage).toBeVisible();
  });

  it("should validate password", async () => {
    const { user } = setup(
      <BrowserRouter>
        <CompanyLoginPage />
      </BrowserRouter>
    );

    const passwordInput = screen.getByLabelText("Password");
    await user.click(passwordInput);
    await user.keyboard("pass");
    const submitButton = screen.getByText("Sign In");
    await user.click(submitButton);
    const errorMessage = screen.getByText(
      "Password cannot be less than 6 characters"
    );
    expect(errorMessage).toBeVisible();
  });
});
