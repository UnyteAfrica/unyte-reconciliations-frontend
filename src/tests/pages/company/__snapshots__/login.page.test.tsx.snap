// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Company Login Page > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center items-center  bg-[#f5f5f5] min-h-screen"
      >
        <div
          class="p-10 bg-white flex justify-center items-center w-[700px] h-[700px] rounded-2xl"
        >
          <form
            class="text-[#333] w-[560px]"
          >
            <header>
              <p
                class="text-center mb-11 font-medium text-[28px]"
              >
                Company Login
              </p>
              <p
                class="mb-11 text-xl"
              >
                Sign in to your account.
              </p>
            </header>
            <div
              class="space-y-6"
            >
              <div
                class=""
              >
                <label
                  class="text-lg text-[#333] leading-[24px] mb-2 block"
                  for="adminEmail"
                >
                  Admin Email
                   
                </label>
                <input
                  class="block text-lg font-sora w-full px-3 py-[16px] h-[72px] border border-[#424242] rounded-lg outline-none focus:border-black"
                  id="adminEmail"
                  name="email"
                  placeholder="johndoe@gmail.com"
                />
                <p
                  class="text-sm text-[#333]"
                />
              </div>
              <div>
                <label
                  class="text-lg text-[#333] leading-[24px] mb-2 block"
                  for="password"
                >
                  Password
                </label>
                <div
                  class="flex relative items-center w-[100%] border border-[#424242] rounded-lg transition duration-300 focus-within:border-black"
                >
                  <div
                    class="w-full"
                  >
                    <input
                      class="block text-lg font-sora px-3 py-[16px] h-[72px] border-[#424242] rounded-lg outline-none mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0"
                      id="password"
                      name="password"
                      placeholder="******"
                      type="password"
                      value=""
                    />
                    <p
                      class="text-sm text-[#333]"
                    />
                  </div>
                  <button
                    class="absolute top-[50%] -translate-y-[50%] right-3"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                      />
                    </svg>
                  </button>
                </div>
                <p
                  class="font-sora text-sm text-[#333]"
                />
              </div>
              <div>
                <p
                  class="mb-2"
                >
                  Don't have an account?
                   
                  <a
                    class="text-primary"
                    href="/company/signup"
                  >
                    Sign up
                  </a>
                </p>
                <div
                  class="flex justify-end"
                >
                  <a
                    class="text-primary mb-2 inline-block"
                    href="/company/forgot-password"
                  >
                    Forgotten Password?
                  </a>
                </div>
                <button
                  class="w-full font-medium text-xl leading-[24px] bg-primary h-[72px] text-white rounded-2xl"
                >
                  Sign In
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center items-center  bg-[#f5f5f5] min-h-screen"
    >
      <div
        class="p-10 bg-white flex justify-center items-center w-[700px] h-[700px] rounded-2xl"
      >
        <form
          class="text-[#333] w-[560px]"
        >
          <header>
            <p
              class="text-center mb-11 font-medium text-[28px]"
            >
              Company Login
            </p>
            <p
              class="mb-11 text-xl"
            >
              Sign in to your account.
            </p>
          </header>
          <div
            class="space-y-6"
          >
            <div
              class=""
            >
              <label
                class="text-lg text-[#333] leading-[24px] mb-2 block"
                for="adminEmail"
              >
                Admin Email
                 
              </label>
              <input
                class="block text-lg font-sora w-full px-3 py-[16px] h-[72px] border border-[#424242] rounded-lg outline-none focus:border-black"
                id="adminEmail"
                name="email"
                placeholder="johndoe@gmail.com"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div>
              <label
                class="text-lg text-[#333] leading-[24px] mb-2 block"
                for="password"
              >
                Password
              </label>
              <div
                class="flex relative items-center w-[100%] border border-[#424242] rounded-lg transition duration-300 focus-within:border-black"
              >
                <div
                  class="w-full"
                >
                  <input
                    class="block text-lg font-sora px-3 py-[16px] h-[72px] border-[#424242] rounded-lg outline-none mb-0 border-0 w-[90%] focus:border-transparent focus:border-b-0"
                    id="password"
                    name="password"
                    placeholder="******"
                    type="password"
                    value=""
                  />
                  <p
                    class="text-sm text-[#333]"
                  />
                </div>
                <button
                  class="absolute top-[50%] -translate-y-[50%] right-3"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M288 144a110.94 110.94 0 0 0-31.24 5 55.4 55.4 0 0 1 7.24 27 56 56 0 0 1-56 56 55.4 55.4 0 0 1-27-7.24A111.71 111.71 0 1 0 288 144zm284.52 97.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400c-98.65 0-189.09-55-237.93-144C98.91 167 189.34 112 288 112s189.09 55 237.93 144C477.1 345 386.66 400 288 400z"
                    />
                  </svg>
                </button>
              </div>
              <p
                class="font-sora text-sm text-[#333]"
              />
            </div>
            <div>
              <p
                class="mb-2"
              >
                Don't have an account?
                 
                <a
                  class="text-primary"
                  href="/company/signup"
                >
                  Sign up
                </a>
              </p>
              <div
                class="flex justify-end"
              >
                <a
                  class="text-primary mb-2 inline-block"
                  href="/company/forgot-password"
                >
                  Forgotten Password?
                </a>
              </div>
              <button
                class="w-full font-medium text-xl leading-[24px] bg-primary h-[72px] text-white rounded-2xl"
              >
                Sign In
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
